### PR TITLE
try to help debug weird error by better testing

### PR DESCRIPTION
**Pull request by sjpotter**
reorder and add some tests to help diagnose error in future

we had a weird error here https://github.com/sjpotter/redisraft/runs/5648599700?check_suite_focus=true

that i've only seen once (and we've run this test many times), but as it should never fail, a bit concerned, so tried to improve the test a bit to help diagnose it.

### CODE REVIEW

**Review by tezc — DISMISSED** [review]

**Comment by tezc** [comment]
@sjpotter  I saw a failure on my local, opened this issue: https://github.com/RedisLabs/redisraft/issues/302

**Comment by sjpotter** [comment]
as your case was different, I did a more intensive change and to add debug logs and turn on debugging in their pytests

File: tests/integration/test_random.py [hunk 1]
```diff
@@ -20,6 +20,8 @@ def test_hash_deterministic_order(cluster):
     """
 
     cluster.create(3)
+    for i in [1, 2, 3]:
+        cluster.node(i).raft_config_set('loglevel', 'debug')
```
Comment (tezc) [inline tests/integration/test_random.py:24]: Don't we already use "debug" level in tests?

**Review by tezc — APPROVED** [review]

File: tests/integration/test_random.py (hunk 1 above)
Comment (sjpotter) [reply tests/integration/test_random.py:24]: wasn't sure, wanted to be sure, as this isn't common, but you might be right.  doesn't hurt as is, just redundant at worst.

Comment (tezc) [reply tests/integration/test_random.py:24]: okay

**Pull request merged**